tests/test_storage.py

```diff
@@ -40,25 +40,41 @@ def test_metadata_target(tmp_metadata_target):
 
 
 @pytest.mark.parametrize(
-    "file_paths", [lazy_fixture("netcdf_local_paths"), lazy_fixture("netcdf_http_paths")]
+    "file_paths",
+    [
+        lazy_fixture("netcdf_local_paths"),
+        lazy_fixture("netcdf_http_paths"),
+        lazy_fixture("netcdf_http_paths_with_secrets"),
+    ],
 )
 @pytest.mark.parametrize("copy_to_local", [False, True])
 @pytest.mark.parametrize("use_cache, cache_first", [(False, False), (True, False), (True, True)])
 @pytest.mark.parametrize("use_dask", [True, False])
 @pytest.mark.parametrize("use_xarray", [True, False])
+@pytest.mark.parametrize("use_query_string_secrets", [True, False])
 def test_file_opener(
-    file_paths, tmp_cache, copy_to_local, use_cache, cache_first, dask_cluster, use_dask, use_xarray
+    file_paths,
+    tmp_cache,
+    fake_secrets,
+    copy_to_local,
+    use_cache,
+    cache_first,
+    dask_cluster,
+    use_dask,
+    use_xarray,
+    use_query_string_secrets,
 ):
     all_paths, _ = file_paths
     path = str(all_paths[0])
     cache = tmp_cache if use_cache else None
+    secrets = fake_secrets if use_query_string_secrets and file_paths[0][-1] == "?" else None
```

Comment:
(I forgot to submit this comment in my review.)

I think `file_paths[0][-1] == "?"` will always evaluate to False, because the last character of the filename is never `?`.

Reply:
This confusion *should* ideally be resolved by https://github.com/pangeo-forge/pangeo-forge-recipes/pull/167/commits/c42151e6ddce7123e4349748ee54b059f9fb8fb0